### DISCUSSION TITLE
[RFC] Responsive Circle Progress

### DISCUSSION THREAD

**Discussion by MadCcc** [ant-design/ant-design#38226] [RFCs - archive]
## Summary
Progress 组件 type `circle` 支持响应式。

## Motivation
在这篇 [RFC](https://github.com/ant-design/ant-design/discussions/38119) 中我们讨论了 Progress 的新型变体，最终得出的结论是这个变体可由当前的 `circle` 类型实现，但需要实现响应式的部分功能。

## API
无变化

## Basic Example
![image](https://user-images.githubusercontent.com/27722486/196707692-5ee98623-9814-42e1-8d35-ff8dbf97c5be.png)

```tsx
import React, { useRef } from React;
import { Progress } from 'antd';

export default () => (
  <Progress type="circle" width={14} format={(percent) => `进行中，已完成${percent}%`} />
);
```

### 进度信息
在上述例子中，进度信息在 Tooltip 中展示，对应 prop 为 Progress 公用的 `format`，hover 热区取消。

当 `width` 小于 20 时，`format` 转为内建 Tooltip 实现，strokeWidth 最小值为 3px

## Drawbacks
No obvious drawbacks yet.

**Comment by li-jia-nan** [comment]
- #38231
